src/main/java/org/moltimate/moltimatebackend/dto/request/DockingRequest.java

```diff
@@ -3,18 +3,46 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
-import org.moltimate.moltimatebackend.model.ligand.PDBQT;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 public class DockingRequest {
-	private PDBQT receptor;
-	private PDBQT ligand;
-	private double centerX;
-	private double centerY;
-	private double centerZ;
-	private double sizeX;
-	private double sizeY;
-	private double sizeZ;
+	private MultipartFile macromolecule;
+	private MultipartFile ligand;
+	private double center_x;
+	private double center_y;
+	private double center_z;
+	private double size_x;
+	private double size_y;
+	private double size_z;
+
+	public MultiValueMap<String, Object> toMap() throws IOException {
+		MultiValueMap<String, Object> map = new LinkedMultiValueMap<>();
+		map.add("macromolecule", new ByteArrayResource( macromolecule.getBytes() ){
+			@Override
+			public String getFilename() {
+				return macromolecule.getOriginalFilename();
+			}
+		});
+		map.add("ligend", new ByteArrayResource( ligand.getBytes() ){//TODO: Fix spelling when API is fixed
```

Comment:
I know that I like to spell ligand wrong but we should probably commit code with the correct spelling :P

Reply:
Spelling has been corrected.